posts/2022-07-07-beta-release-post.adoc

```diff
@@ -0,0 +1,104 @@
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+---
+= Enhance the effectiveness of log analysis tools with new JSON fields for logged exceptions in Open Liberty 22.0.0.8-beta
+Ryan Storey <https://github.com/ryan-storey>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+//Blank line here is necessary before starting the body of the post.
+
+Open Liberty 22.0.0.8 offers new JSON fields for logged exceptions, enhancing the Open Liberty JSON logging and logstashCollector-1.0 features, by increasing the effectiveness of log analysis tools down stream, as rules and visualizations relevant to these two new exception related fields can be created which results in more stream-lined strategy to identify any issues encountered by the application.
+
+The link:{url-about}[Open Liberty] 22.0.0.8-beta includes the following beta features (along with all GA features):
+
+* <<json_fields, New JSON fields for logged exceptions (via parameters)>>
+
+[#json_fields]
+== New JSON fields for logged exceptions (via parameters)
+
+Incorporated as part of the Java's logger API are methods that allow you to include a `Throwable` as a parameter. This new functionality enhances Open Liberty's JSON logging and the link:{url-prefix}/docs/latest/reference/feature/logstashCollector-1.0.html[`logstashCollector-1.0` feature] by introducing two new fields relevant to the `Throwable` object. For Open Liberty's JSON logging, the fields are `ibm_stackTrace` and `ibm_exceptionName`. For the `logstashCollector-1.0` feature the fields are `stackTrace` and `exceptionName`. The `stack trace` field would present solely the stack trace of the `Throwable`. The `exception name` field would present the type of exception of the `Throwable`. These fields are only included if the logging record includes a `Throwable` object. This affects the message and trace events for both the Open Liberty JSON logging and `logstashCollector-1.0` features.
+
+Prior to this enhancement, any logging record originating from a Java Logger which made use of any of the methods that accept a `Throwable` parameter would simply append the stack trace to the existing `message` field.  By keeping the `message` field solely for the logged message and having a unique field for the stack trace and exception type enhances the effectiveness of log analysis tools down stream. Rules and visualizations relevant to these two new exception related fields can be created which results in more stream-lined strategy to identify any issues encountered by the application. No configuration is necessary to enable the new functionality.
```

Comment:
```suggestion
The stack trace is now separated from logged messages in logging records so that log analysis tools can present them more clearly. This makes visualizations of the logs by downstream log analysis tools easier to read when you are identifying any issues encountered by the application. Previously, any logging record originating from a Java Logger object that made use of any of the methods that accept a `Throwable` parameter would simply append the stack trace to the existing `message` field. Keeping the `message` field solely for the logged message and having a separate field for the stack trace and exception type enhances the effectiveness of log analysis tools down stream.

Java's link:https://docs.oracle.com/javase/8/docs/api/java/util/logging/package-summary.html[Logging API] provides methods that allow you to include a `Throwable` as a parameter. When the `Throwable` object is used, Open Liberty's JSON logging provides two new fields, `ibm_stackTrace` and `ibm_exceptionName`, and the Open Liberty link:{url-prefix}/docs/latest/reference/feature/logstashCollector-1.0.html[Logstash Collector] feature provides two new fields, `stackTrace` and `exceptionName`. The stack trace fields present only the stack trace of the `Throwable` object. The exception name fields present the type of exception of the `Throwable` object.
```